### PR TITLE
Fixes + Updates Seasons API

### LINKED ISSUE #217: v4/seasons: Bug causes only some entries to be listed
I've tested all seasons on 2021 and 2022 and this bug happens on all of them

Let's take **Winter 2022** for example:
- https://myanimelist.net/anime/season/2022/winter
- https://api.jikan.moe/v4/seasons/2022/winter?page=1
- https://api.jikan.moe/v4/seasons/2022/winter?page=2

The API is missing most entries, it's only listing type "TV" entries

Examples of Winter 2022 entries not listed, one per category
- Movie: https://myanimelist.net/anime/49310/Fruits_Basket__Prelude
- OVA: https://myanimelist.net/anime/49521/Non_Non_Biyori_Nonstop_OVA
- ONA: https://myanimelist.net/anime/43693/Thermae_Romae_Novae
- Special: https://myanimelist.net/anime/50360/Mushoku_Tensei__Isekai_Ittara_Honki_Dasu_Special

### PR DESCRIPTION
**Pull request by irfan-dahir**
- Fixes `/seasons/*` not returning other entry types
- Adds `type` query filter to `/seasons/*`

Resolves #217 

----
Don't think we need to add any more filter types here. If the user wants to create an advanced seasonal query, they can use the search API instead.

e.g `/v4/seasons/2022/fall?type=movie` corresponds directly to
`/v4/anime?start_date=2022-10-1&end_date=2022-12-31&type=movie`

⚠Note: Need to run `php artisan swagger-lume:generate` once this PR is updated with the latest changes in master from #285

### CODE REVIEW

**Comment by irfan-dahir** [comment]
Since this is the only filter available here, I'm thinking of renaming `type` to `filter`. This would be the same pattern that is followed in other requests like Top. e.g `/top/anime?filter=airing` as it corresponds to the tabs provided on the MAL page. 🤔 

That would make more sense IMO.

**Review by pushrbx — CHANGES_REQUESTED** [review]

**Comment by AniLeo** [comment]
Thanks for the fix, will keep an eye for the merge

**Comment by Green-Cat** [comment]
Any chance to have this merged?

**Comment by irfan-dahir** [comment]
@pushrbx g2g?

**Comment by pushrbx** [comment]
> @pushrbx g2g?

I'll try to review it again and test it in the following week. 👍  (Nov 21-25)

File: app/Http/Controllers/V4DB/SeasonController.php [hunk 1]
```diff
@@ -302,4 +337,30 @@ private function getSeasonStr() : array
...
+
+        return [
+            'from' => Carbon::createFromDate($year, $monthStart, 1)->toAtomString(),
+            'to' => Carbon::createFromDate($year, $monthEnd, 1)->modify('last day of this month')->toAtomString()
+        ];
+
+//        return [
+//            'from' => (new \DateTime())->setDate($year, $monthStart, 1)->format(\DateTimeInterface::ATOM),
+//            'to' => (new \DateTime())->setDate($year, $monthEnd, 1)->modify('last day of this month')->format(\DateTimeInterface::ATOM)
+//        ];
```
Comment (pushrbx) [inline app/Http/Controllers/V4DB/SeasonController.php:364]: commented out code is frowned upon! 😦

**Review by pushrbx — CHANGES_REQUESTED** [review]

**Comment by pushrbx** [comment]
> ⚠Note: Need to run php artisan swagger-lume:generate once this PR is updated with the latest changes in master

This can be part of a github action too

**Review by pushrbx — APPROVED** [review]
Charlie Day Thumb

**Pull request merged**